source/common/common/logger.h

```diff
@@ -192,7 +207,10 @@ template <Id id> class Loggable {
 /**
  * Convenience macro to log to a user-specified logger.
  */
-#define ENVOY_LOG_TO_LOGGER(LOGGER, LEVEL, ...) ENVOY_LOG_##LEVEL##_TO_LOGGER(LOGGER, ##__VA_ARGS__)
+#define ENVOY_LOG_TO_LOGGER(LOGGER, LEVEL, ...)                                                    \
```

Comment:
Would you mind adding a test for this?  I think this may suffer from the not-a-statement macro issue (see [here](https://stackoverflow.com/questions/154136/why-use-apparently-meaningless-do-while-and-if-else-statements-in-macros)), but I'm not sure.  Specifically, I'm worried about the case:
```
if (condition) ENVOY_LOG_TO_LOGGER(LOGGER, LEVEL, ...);
else f();
```
This may be the reason that the behavior at the callsite in source/common/http/filter/lua/lua_filter.cc needed to be updated.

Reply:
Wrapped macro into do ... while(0) based on the link you sent (thanks!). 
I had to keep changes for source/common/http/filter/lua/lua_filter.cc. 
spdlog::log ... returns void and statement 
return ENVOY_LOG...
was an easy way to log and exit function in one step. It will not work with the new macro.

Reply:
SGTM, thanks!